src/cpr-mask.helpers.js

```diff
@@ -184,23 +186,27 @@ function charMatchesRegexPattern(pattern, char) {
 	if (pattern === "1") return /\d/.test(char);
 	if (pattern === "W") return /[A-Za-z0-9]/.test(char);
 	if (pattern === "*") return /./.test(char);
+	return true;
 }
 
-export function getMaskSlice(pattern, start, stop) {
+export function getMaskSlice(pattern, input, start, stop) {
 	let totalRequired = stop - start;
 	let found = "";
 	let i = start;
 	let selectionMove = 0;
 	while (found.length < totalRequired) {
 		if (/[A1W*]/.test(pattern[i])) {
 			found += pattern[i];
+		} else if (pattern[i] === input[i]) {
+			found += pattern[i];
+			selectionMove++;
```

Comment:
The only reason we need the input here is to make sure the selectionMove gets increased when an input value character includes a pattern character like "-" or "(". Again, this only affects pasting that includes pattern characters.